File: frontend/src/app/dto/Post.ts
import { Comment } from "./Comment";
import { User } from "./User";

export class Post {

  id: number | null = null;
  title:string | null = null;
  body: string | null = null;
  user: User | null = null;
  commentList: Array<Comment> | null = null;
  createdDateTime: string | null = null;  // `yyyy-MM-dd HH:mm:ss

  constructor(o?: Partial<Post>) {
    Object.assign(this, o);

  }
}

File: backend/src/main/java/com/example/backend/entity/Post.java
package com.example.backend.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Post extends AuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    // POST MANY TO ONE USER INFO
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userInfo_id")
    private UserInfo userInfo;

    // POST ONE TO MANY COMMENT
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @OneToMany(mappedBy = "post", fetch = FetchType.LAZY)
    private List<Comment> commentList = new ArrayList<>();
    @Transient
    @JsonProperty(access = JsonProperty.Access.READ_WRITE)
    private List<Comment> commentListSerde = new ArrayList<>();

    private String title;
    private String body;

    public Post(Long id) {
        this.id = id;
    }
}
